client/allocrunner/alloc_runner_hooks.go

```diff
@@ -149,7 +149,7 @@ func (ar *allocRunner) initRunnerHooks(config *clientconfig.Config) error {
 		newCgroupHook(ar.Alloc(), ar.cpusetManager),
 		newUpstreamAllocsHook(hookLogger, ar.prevAllocWatcher),
 		newDiskMigrationHook(hookLogger, ar.prevAllocMigrator, ar.allocDir),
-		newAllocHealthWatcherHook(hookLogger, alloc, hs, ar.Listener(), ar.consulClient, ar.checkStore),
+		newAllocHealthWatcherHook(hookLogger, alloc, builtTaskEnv, hs, ar.Listener(), ar.consulClient, ar.checkStore),
```

Comment:
I think re-using it here is safe since these are read-only? Another option is to call `envBuilder.Build()` in each function that needs a `TaskEnv`.

Reply:
_Narrator: it was not safe._